### PR TITLE
Improve current look of value label in GraphTrack

### PR DESCRIPTION
**Pull request by vickyliu-go4it**
We make the following changes in the value label of `GraphTrack`:
* Change the white value label to be transparent;
* Change the font size to be the same as the legend font size;
* Adjust the position of text in the white value label.

Test: Run orbit with the flag `--devmode` or `--enable_cgroup_memory`, check the value label in the graph tracks.

After the change:
https://screenshot.googleplex.com/8N9csiHWhT7fdDV

Before the change:
https://screenshot.googleplex.com/8z3dn7UM4AYpXc7

### CODE REVIEW

File: src/OrbitGl/GraphTrack.cpp [hunk 1]
```diff
@@ -171,24 +172,39 @@ template <size_t Dimension>
...
     text_width = std::max(text_width, text_renderer.GetStringWidth(line.c_str(), font_size));
   }
-  float text_height = layout_->GetTextBoxHeight() * lines.size();
+  float single_line_height = text_renderer.GetStringHeight(text.c_str(), font_size);
+  float text_height = single_line_height * lines.size() + kSpaceBetweenLines * (lines.size() - 1);
   Vec2 text_box_size(text_width, text_height);
 
   float arrow_width = text_box_size[1] / 2.f;
   bool arrow_is_left_directed =
       target_pos[0] < viewport_->GetWorldTopLeft()[0] + text_box_size[0] + arrow_width;
```
Comment (antonrohr) [inline src/OrbitGl/GraphTrack.cpp:193]: I think this now also needs the margins.

**Review by antonrohr — APPROVED** [review]
Very nice change! I like how it looks

File: src/OrbitGl/GraphTrack.cpp [hunk 2]
```diff
@@ -71,12 +66,18 @@ void GraphTrack<Dimension>::Draw(Batcher& batcher, TextRenderer& text_renderer,
   float point_x = time_graph_->GetWorldFromTick(label_time);
   float point_y = GetLabelYFromValues(values);
   std::string text = GetLabelTextFromValues(values);
-  DrawLabel(batcher, text_renderer, Vec2(point_x, point_y), text, kBlack, kWhite, label_z);
+  const Color kLabelTextColor(0, 0, 0, 255);
+  const Color kLabelBackgroundColor(255, 255, 255, 100);
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:70]: Could I ask why we change the name of the colors? I rather would prefer to have kTransparentWhite and kBlack

File: src/OrbitGl/GraphTrack.cpp [hunk 3]
```diff
@@ -71,12 +66,18 @@ void GraphTrack<Dimension>::Draw(Batcher& batcher, TextRenderer& text_renderer,
...
+  float label_z = GlCanvas::kZValueTrackLabel + z_offset;
+  DrawLabel(batcher, text_renderer, Vec2(point_x, point_y), text, kLabelTextColor,
+            kLabelBackgroundColor, label_z);
 
   if (Dimension == 1) return;
 
   // Draw legends
-  DrawLegend(batcher, text_renderer, series_.GetSeriesNames(), kWhite, text_z);
+  const Color kLegendTextColor(255, 255, 255, 255);
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:78]: and kWhite

File: src/OrbitGl/GraphTrack.cpp [hunk 4]
```diff
@@ -171,24 +172,39 @@ template <size_t Dimension>
...
   bool arrow_is_left_directed =
       target_pos[0] < viewport_->GetWorldTopLeft()[0] + text_box_size[0] + arrow_width;
   Vec2 text_box_position(
-      target_pos[0] + (arrow_is_left_directed ? arrow_width : -arrow_width - text_box_size[0]),
+      target_pos[0] + (arrow_is_left_directed ? arrow_width + kTextLeftMargin
+                                              : -arrow_width - kTextRightMargin - text_box_size[0]),
       target_pos[1] - text_box_size[1] / 2.f);
 
-  Box arrow_text_box(text_box_position, text_box_size, z);
+  float bottom_y_of_first_line = text_box_position[1] + text_box_size[1] - single_line_height;
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:199]: which is the first line? the one at the top or the one at the bottom?

File: src/OrbitGl/GraphTrack.cpp (hunk 2 above)
Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp]: Changed back to use kBlack and kTransparentWhite. Thanks!

File: src/OrbitGl/GraphTrack.cpp (hunk 3 above)
Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp]: Done! Thanks!

File: src/OrbitGl/GraphTrack.cpp (hunk 4 above)
Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp]: Thanks for pointing out this misleading name. Changed to use `bottom_y_of_top_line`. Thanks!

File: src/OrbitGl/GraphTrack.cpp [hunk 5]
```diff
@@ -171,24 +172,39 @@ template <size_t Dimension>
...
+      target_pos[0] < viewport_->GetWorldTopLeft()[0] + arrow_box_size[0] + arrow_width;
   Vec2 text_box_position(
-      target_pos[0] + (arrow_is_left_directed ? arrow_width : -arrow_width - text_box_size[0]),
+      target_pos[0] + (arrow_is_left_directed ? arrow_width + kTextLeftMargin
+                                              : -arrow_width - kTextRightMargin - text_box_size[0]),
       target_pos[1] - text_box_size[1] / 2.f);
 
-  Box arrow_text_box(text_box_position, text_box_size, z);
+  float bottom_y_of_top_line = text_box_position[1] + text_box_size[1] - single_line_height;
+  text_renderer.AddText(text.c_str(), text_box_position[0], bottom_y_of_top_line, z, text_color,
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:202]: So, when writing a multi-line string we have to provide the bottom_y of the top_line? So strange!

**Review by karupayun — APPROVED** [review]
LGTM! The arrow boxes looks much more nicer now! :)

File: src/OrbitGl/GraphTrack.cpp (hunk 1 above)
Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp]: Done! Thanks!

File: src/OrbitGl/GraphTrack.cpp (hunk 5 above)
Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp:202]: I guess we only need to specify the location to add the top line in order to add a multiline string. And for the opengl coordinate system, we need to specify the position of the bottom left point.

**Pull request merged**